### PR TITLE
FINERACT-2023: Improve contributor's guide

### PR DESCRIPTION
**Pull request by abhinav7sinha**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-doc/src/docs/en/chapters/contributing/contributing.adoc [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+== Dev environment setup:
+
+Follow link:../development/dev-env-setup.adoc[these] steps to set up your dev environment.
+
+== Identify something to work on
+
+* Browse existing link:https://issues.apache.org/jira/browse/FINERACT-2055?jql=project%20%3D%20FINERACT%20ORDER%20BY%20created%20DESC[issues]
+** Filter by tags - "volunteer", "gsoc", "new contributor", etc.
+* Browse link:https://mifosforge.jira.com/wiki/spaces/RES/pages/3532095546/Google+Summer+of+Code+2024+Ideas[GSoC projects]
```
Comment (jdailey) [inline fineract-doc/src/docs/en/chapters/contributing/contributing.adoc]: We cannot point to an external project

File: fineract-doc/src/docs/en/chapters/demos/video-demonstration.adoc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+= VIDEO DEMONSTRATION
+
+Apache Fineract / Mifos X Demo (November 2016) - <https://www.youtube.com/watch?v=h61g9TptMBo>
```
Comment (jdailey) [inline fineract-doc/src/docs/en/chapters/demos/video-demonstration.adoc]: This is really old.  Let's delete or create a new one.  I think  it should be a demo of the APIs

**Review by jdailey — COMMENTED** [review]
A few changes I think are needed.

**Comment by abhinav7sinha** [comment]
> A few changes I think are needed.

agreed. I've made the changes now

File: fineract-doc/src/docs/en/chapters/demos/video-demonstration.adoc (hunk 2 above)
Comment (abhinav7sinha) [reply fineract-doc/src/docs/en/chapters/demos/video-demonstration.adoc]: @jdailey Done.

File: fineract-doc/src/docs/en/chapters/contributing/contributing.adoc (hunk 1 above)
Comment (abhinav7sinha) [reply fineract-doc/src/docs/en/chapters/contributing/contributing.adoc]: Done.

**Comment by adamsaghy** [comment]
@abhinav7sinha Please fix the conflicts (rebase your PR) and squash the commits!

**Comment by jdailey** [comment]
Is this stale?   What changes are needed to accept this?

**Comment by Napho** [comment]
@abhinav7sinha is this something that you are still open to finishing, I am looking to make contributions to the documentation mainly finishing the TBD sections and i think a lot of the work is already done here. If you are open to someone else taking over these changes and getting them in please let me know

**Pull request closed without merging**